src/main/groovy/com/spidasoftware/schema/conversion/changeset/v7/RelativeElevationChangeSet.groovy

```diff
@@ -0,0 +1,105 @@
+/*
+ * ©2009-2020 SPIDAWEB LLC
+ */
+package com.spidasoftware.schema.conversion.changeset.v7
+
+import com.spidasoftware.schema.conversion.changeset.ConversionException
+import com.spidasoftware.schema.conversion.changeset.calc.AbstractCalcDesignChangeset
+
+import static java.lang.Math.sin
+
+/**
+ * v7.3 adds relative elevation to wireEndPoints. Removes "relative elevation" when we down convert.
+ */
+class RelativeElevationChangeSet extends AbstractCalcDesignChangeset {
+
+	protected double footMetreSiConversion = 0.3048d
+	protected double radianDegreeSiConversion = 57.29577951308232d
+
+	double convertToMetre(double foot) {
+		return foot * footMetreSiConversion
+	}
+
+	double convertToFoot(double metre) {
+		return metre / footMetreSiConversion
+	}
+
+	double convertToRadian(double degree) {
+		return degree / radianDegreeSiConversion
+	}
+
+	double calculateInclination(double relativeElevation, double distance) {
+		return Math.asin(relativeElevation/distance as double)
+	}
+
+	double calculateRelativeElevation(double inclinationRad, double distance) {
+		double sinInclination = sin(inclinationRad)
+		return distance * sinInclination
+	}
+
+	@Override
+	void applyToDesign(Map designJSON) throws ConversionException {
+		def wireEndPoints = designJSON.structure?.wireEndPoints
+		wireEndPoints.each { Map wireEndPoint ->
+
+			def inclinationMap = wireEndPoint.get("inclination") as Map
+			def distanceMap = wireEndPoint.get("distance") as Map
+			if (distanceMap != null && inclinationMap != null) {
+
+				double inclination = inclinationMap.get("value") as double
+				double distance = distanceMap.get("value") as double
+
+				if (inclinationMap.get("unit") == "DEGREE_ANGLE") {
+					inclination = convertToRadian(inclination)
+				}
+
+				boolean convertedFromFoot = false
+				if (distanceMap.get("unit") == "FOOT") {
+					distance = convertToMetre(distance)
+					convertedFromFoot = true
+				}
+
+				double relativeElevation = calculateRelativeElevation(inclination, distance)
+				if (convertedFromFoot) { //if distance was saved in foot then insert relativeElevation in foot
```

Comment:
We will always output in SI and store in SI in SPIDAdb. So I would lose all of the converting to feet and always do the conversion and output it in meters.